WeatherApp/ViewController.swift

```diff
@@ -8,10 +8,72 @@
 import UIKit
 
 class ViewController: UIViewController {
+    
+    private var weatherViewModel: WeatherViewModel!
+
+    lazy var cityLabel: UILabel = {
+        let city = UILabel()
+        city.textColor = .white
+        city.translatesAutoresizingMaskIntoConstraints = false
+        return city
+    }()
+    
+    lazy var temperatureLabel: UILabel = {
+        let temperature = UILabel()
+        temperature.font = UIFont.systemFont(ofSize: 70.0)
+        temperature.textColor = .white
+        temperature.translatesAutoresizingMaskIntoConstraints = false
+        return temperature
+    }()
+    
+    lazy var descriptionLabel: UILabel = {
+        let description = UILabel()
+        description.textColor = .white
+        description.translatesAutoresizingMaskIntoConstraints = false
+        return description
+    }()
+    
+    lazy var vstack: UIStackView = {
+        let stack = UIStackView(arrangedSubviews: [cityLabel, temperatureLabel, descriptionLabel])
+        stack.axis = .vertical
+        stack.alignment = .center
+        stack.translatesAutoresizingMaskIntoConstraints = false
+        return stack
+    }()
+    
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        setUpViews()
+        setUpConstraints()
+        callToViewModelForUIUpdate()
+    }
+    
+    private func callToViewModelForUIUpdate() {
+        self.weatherViewModel = WeatherViewModel()
+        self.weatherViewModel.bindWeatherViewModelToController = {
+            self.configure()
+        }
+    }
+    
+    private func configure() {
+        DispatchQueue.main.async {
+            self.cityLabel.text = self.weatherViewModel.weatherData.name
```

Comment:
Los datos de una classe son suyos y no los debes de exponer... violas el principio de programacion: "Don't talk to strangers" o lei de Demeter. Para eso en el `weatherViewModel` encapsula tus datos y expone solo lo que necesitas.
```suggestion
            cityLabel.text = weatherViewModel.name
```
Puedes usar computed variables para llamar `name` directamente. La keyword `self` no se usa... solo cuando es necesario especificar cual de las variables que tenen el mismo nombre es la interna, que suele ser lo que pasa en los constructores.